code/modules/nano/modules/camera_monitor.dm

```diff
@@ -0,0 +1,188 @@
+			if(!can_access_camera(C))
+				continue
+
+			var/cam = C.nano_structure()
+			cameras[++cameras.len] = cam
+
+		camera_cache=list2json(cameras)
+
+	tempnets.Cut()
+	access = list(access_captain) // Assume captain level access when AI
+
+	// Loop through the ID's permission, and check which networks the ID has access to.
+	for(var/l in networks) // Loop through networks.
+		for(var/m in networks[l]) // Loop through access levels of the networks.
+			if(m in access)
+				if(l in network) // Checks if the network is currently active.
+					tempnets.Add(list(list("name" = l, "active" = 1)))
+				else
+					tempnets.Add(list(list("name" = l, "active" = 0)))
+				break
+	if(tempnets.len)
+		data["networks"] = tempnets
+
+	if(current)
+		data["current"] = current.nano_structure()
+	data["cameras"] = list("__json_cache" = camera_cache)
+
+	ui = nanomanager.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if (!ui)
+		ui = new(user, src, ui_key, "sec_camera.tmpl", "Camera Console", 900, 800)
+
+		// adding a template with the key "mapContent" enables the map ui functionality
+		ui.add_template("mapContent", "sec_camera_map_content.tmpl")
+		// adding a template with the key "mapHeader" replaces the map header content
+		ui.add_template("mapHeader", "sec_camera_map_header.tmpl")
+
+		ui.set_initial_data(data)
+		ui.open()
+		ui.set_auto_update(1)
+		
+/datum/nano_module/camera_monitor/Topic(href, href_list)
+	if(..()) 
+		return 1
+
+	if(href_list["switchTo"])
+		if(usr.stat || ((get_dist(usr, src) > 1 || !( usr.canmove ) || usr.blinded) && !istype(usr, /mob/living/silicon))) return
+		var/obj/machinery/camera/C = locate(href_list["switchTo"]) in cameranet.cameras
+		if(!C) return
+
+		switch_to_camera(usr, C)
+		return 1
+	else if(href_list["reset"])
+		if(usr.stat || ((get_dist(usr, src) > 1 || !( usr.canmove ) || usr.blinded) && !istype(usr, /mob/living/silicon))) return
+		reset_current()
+		usr.check_eye(current)
+		return 1
+	else if(href_list["activate"]) // Activate: enable or disable networks
+		var/net = href_list["activate"]	// Network to be enabled or disabled.
+		var/active = href_list["active"] // Is the network currently active.
+		for(var/a in networks[net])
+			if(a in access) // Re-check for authorization.
+				if(text2num(active) == 1)
+					src.network -= net
+					break
+				else
+					src.network += net
+					break
+		invalidateCameraCache()
+		nanomanager.update_uis(src)
+
+
+// Check if camera is accessible when jumping
+/datum/nano_module/camera_monitor/proc/can_access_camera(var/obj/machinery/camera/C)
+	var/list/shared_networks = src.network & C.network
+	if(shared_networks.len)
+		return 1
+	return 0
+
+// Switching to cameras
+/datum/nano_module/camera_monitor/proc/switch_to_camera(var/mob/user, var/obj/machinery/camera/C)
+	if ((get_dist(user, src) > 1 || user.machine != src || user.blinded || !( user.canmove ) || !( C.can_use() )) && (!istype(user, /mob/living/silicon/ai)))
```

Comment:
The last clause of this line - wouldn't it be better suited for an `isAI()` check?